### ISSUE TITLE
Implement Kernel Mean Matching

### ISSUE THREAD

**Issue by antoinedemathelin** [scikit-adaptation/skada#82]
Hi everyone,
I am glad to see that the skada repo is already public. I really like the API choices that have been made (the pipeline idea is great!)
I am opening this issue to propose the implementation of the Kernel Mean Matching reweighting method (cf. “Correcting sample selection bias by unlabeled data.” [paper](https://proceedings.neurips.cc/paper_files/paper/2006/file/a2186aa7c086b46ad4e8bf81e2a3a19b-Paper.pdf).

Are you ok to add it to the library ? If yes, I can open a PR.

**Comment by rflamary** [comment]
Hello @antoinedemathelin and welcome back to skada !. You have already coded many methods that are of interest to us.

This is a very good idea, KMM is a reweighting isn't it? you could do it with a specific Adapter that outputs sample_weights ;)

**Comment by antoinedemathelin** [comment]
Thank you @rflamary !
Yes it's a reweighting algorithm. From what I understand, I should write a `KMMAdapter` object, implementing an `adapt` method which returns an `AdaptationOutput` instance containing the sample_weights of the reweighting. If I am right, the inputs arrays `X, y`, now contain both source and target samples, so the `sample_weights` array should contain the source importance weights for the source indexes and 0 for the target indexes ? I will follow the `KLIEP` skada implementation, the main difference with KMM is the optimization part, the "API part" should be the same, I guess.

I guess, I need to write a small example and tests for the method ?

**Comment by tgnassou** [comment]
Exactly! If you have any problem, just ask us. In the example don't hesitate to show how works the method with nice plots ;) 

And don't hesitate to give us some feedback on the API, we are still converging to the final API
